frontend/src/lib/components/apps/components/layout/AppDecisionTree.svelte

```diff
@@ -151,6 +151,19 @@
 			subGridIndex: nodes.findIndex((node) => node.id === currentNodeId)
 		}
 	}
+
+	$: {
+		const nodeIds = nodes.map((node) => node.id)
+		const debuggingComponentsIds = Object.keys($debuggingComponents)
+
+		if (JSON.stringify(nodeIds) !== JSON.stringify(debuggingComponentsIds)) {
+			debuggingComponentsIds.forEach((id) => {
+				if (!nodeIds.includes(id)) {
+					delete $debuggingComponents[id]
+				}
+			})
+		}
+	}
```

Comment:
reactive statement should be a function call with just the parameters to watch otherwise this will be called way too much